### PR TITLE
Allow setting custom Puppeteer launch options

### PR DESCRIPTION
**Pull request by timoschwarzer**
Depends on https://github.com/sindresorhus/capture-website/pull/9

### Todo:
1. Wait for https://github.com/sindresorhus/capture-website/pull/9 to be merged.
2. Update version dependency of `capture-website` in `package.json`.

### CODE REVIEW

File: cli.js [hunk 1]
```diff
@@ -139,6 +140,7 @@ const options = cli.flags;
 options.modules = arrify(options.module);
 options.scripts = arrify(options.script);
 options.styles = arrify(options.style);
+options.launchOptions = JSON.parse(options.launchOptions || 'null') || {};
```
Comment (sindresorhus) [inline cli.js]:
```suggestion

if (options.launchOptions) {
	options.launchOptions = JSON.parse(options.launchOptions);
}
```

**Comment by timoschwarzer** [comment]
I'll come back to this tomorrow! :+1:

**Comment by sindresorhus** [comment]
Bump :)

**Comment by timoschwarzer** [comment]
Oh yeah, totally forgot about this. I was ill the last week and had to catch up some other work... Give me some minutes :)

**Comment by timoschwarzer** [comment]
Rebased. I'm not sure how to generate the AVA snapshot and report though. If you could give me a small hint... :)

**Comment by sindresorhus** [comment]
`$ ava -u`

**Comment by sindresorhus** [comment]
Or `$ npx ava -u` if you don't have AVA installed globally.

File: cli.js [hunk 2]
```diff
@@ -36,6 +36,7 @@ const cli = meow(`
 	  --authentication         Credentials for HTTP authentication
 	  --debug                  Show the browser window to see what it's doing
 	  --overwrite              Overwrite the destination file if it exists
+	  --launch-options         JSON formatted puppeteer launch options
```
Comment (sindresorhus) [inline cli.js]:
```suggestion
	  --launch-options         Puppeteer launch options as JSON
```

**Comment by timoschwarzer** [comment]
Should be fine now. I had to modify the test script a bit to allow escaped double quotes inside arguments.

File: cli.js [hunk 3]
```diff
@@ -36,6 +36,7 @@ const cli = meow(`
 	  --authentication         Credentials for HTTP authentication
 	  --debug                  Show the browser window to see what it's doing
 	  --overwrite              Overwrite the destination file if it exists
+	  --launch-options         Puppeteer launch options as JSON
```
Comment (sindresorhus) [inline cli.js]: This needs to be reflected in the readme too.

File: cli.js [hunk 4]
```diff
@@ -62,6 +63,7 @@ const cli = meow(`
 	  --user-agent="I love unicorns"
 	  --cookie="id=unicorn; Expires=Wed, 21 Oct 2018 07:28:00 GMT;"
 	  --authentication="username:password"
+	  --launch-options='{ "headless": false }'
```
Comment (sindresorhus) [inline cli.js]:
```suggestion
	  --launch-options='{"headless": false}'
```

File: test.js [hunk 5]
```diff
@@ -40,9 +40,13 @@ test('check flags', async t => {
 --user-agent="I love unicorns"
 --cookie="id=unicorn; Expires=Wed, 21 Oct 2018 07:28:00 GMT;"
 --authentication="username:password"
+--launch-options="{ \\"headless\\": false }"
```
Comment (sindresorhus) [inline test.js]:
```suggestion
--launch-options="{\\"headless\\": false}"
```

File: test.js.md [hunk 6]
```diff
@@ -9,52 +9,60 @@ Generated by [AVA](https://ava.li).
...
       debug: false,
       delay: 10,
-      element: '.main-content',
-      emulateDevice: 'iPhone X',
-      fullPage: false,
-      header: 'x-powered-by: capture-website-cli',
-      headers: {
-        'x-powered-by': 'capture-website-cli',
+      element: {
+        'main-conten': true,
```
Comment (sindresorhus) [inline test.js.md]: This is not correct

Comment (timoschwarzer) [reply test.js.md]: Uhm, I just launched `ava -u`

File: cli.js (hunk 3 above)
Comment (timoschwarzer) [reply cli.js]: Done

File: test.js.md (hunk 6 above)
Comment (sindresorhus) [reply test.js.md]: There's clearly something incorrect about your replacement thing: https://github.com/sindresorhus/capture-website-cli/pull/2/files#diff-1dd241c4cd3fd1dd89c570cee98b79ddR47

`main-conten` is missing a `t`, for example.

Comment (timoschwarzer) [reply test.js.md]: Fixed.

**Pull request merged**